rest_framework/serializers.py

```diff
@@ -947,6 +947,11 @@ def get_fields(self):
         # Determine any extra field arguments and hidden fields that
         # should be included
         extra_kwargs = self.get_extra_kwargs()
+        for field in extra_kwargs:
+            assert field not in declared_fields, (
+                "Field {} is declared on the class and also present in "
+                "extra_kwargs or read_only_fields".format(field)
+            )
```

Comment:
On Wed, Jun 15, 2016 at 5:56 PM, Tom Christie <notifications@github.com>
wrote:

> In rest_framework/serializers.py
> <https://github.com/tomchristie/django-rest-framework/pull/4199#discussion_r67177845>
> :
>
> > @@ -947,6 +947,11 @@ def get_fields(self):
> >          # Determine any extra field arguments and hidden fields that
> >          # should be included
> >          extra_kwargs = self.get_extra_kwargs()
> > +        for field in extra_kwargs:
> > +            assert field not in declared_fields, (
> > +                "Field {} is declared on the class and also present in "
> > +                "extra_kwargs or read_only_fields".format(field)
> > +            )
>
> On thinking about this: we *might* need to be a bit clever about this -
> some folks may want to use subclassing to override a declared field.
> Unclear.
>
>
> Can you give me an example, and what you think the correct behaviour
should be?

-- 
George-Cristian Bîrzan

On Wed, Jun 15, 2016 at 5:56 PM, Tom Christie <notifications@github.com>
wrote:

> In rest_framework/serializers.py
> <https://github.com/tomchristie/django-rest-framework/pull/4199#discussion_r67177845>
> :
>
> > @@ -947,6 +947,11 @@ def get_fields(self):
> >          # Determine any extra field arguments and hidden fields that
> >          # should be included
> >          extra_kwargs = self.get_extra_kwargs()
> > +        for field in extra_kwargs:
> > +            assert field not in declared_fields, (
> > +                "Field {} is declared on the class and also present in "
> > +                "extra_kwargs or read_only_fields".format(field)
> > +            )
>
> On thinking about this: we *might* need to be a bit clever about this -
> some folks may want to use subclassing to override a declared field.
> Unclear.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/tomchristie/django-rest-framework/pull/4199/files/5dd336ec6ca80dbcabaa1be054ac5a7f2f6c2a7d#r67177845>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAv3ZReOQoFmXw-gCmRa6DrrjKdDisEhks5qMBKigaJpZM4I2QWM>
> .
>



-- 
George-Cristian Bîrzan